src/index/reader.rs

```diff
@@ -110,11 +117,50 @@ impl IndexReader {
         let meta = factory
             .create_segment_metadata_client(scoped_segment.clone())
             .await;
+
+        let controller_client = factory.controller_client();
+        let endpoint = controller_client
+            .get_endpoint_for_segment(&scoped_segment)
+            .await
+            .expect("get endpoint for segment");
+        let raw_client = factory.create_raw_client_for_endpoint(endpoint);
+        let segment_name = scoped_segment.to_string();
+        let token = controller_client
+            .get_or_refresh_delegation_token_for(stream.clone())
+            .await
+            .expect("controller error when refreshing token");
+        let request = Requests::GetSegmentAttribute(GetSegmentAttributeCommand {
+            request_id: get_request_id(),
+            segment_name: segment_name.clone(),
+            attribute_id: INDEX_RECORD_SIZE_ATTRIBUTE_ID,
+            delegation_token: token,
+        });
+        let reply = raw_client
+            .send_request(&request)
+            .await
+            .expect("get segment attribute");
+
+        let record_size = match reply {
+            Replies::SegmentAttribute(cmd) => {
+                if cmd.value == i64::MIN {
+                    info!("Segment attribute for record_size is not set.Falling back to default RECORD_SIZE = {:?}", RECORD_SIZE);
+                    RECORD_SIZE as usize
+                } else {
+                    cmd.value as usize
+                }
+            }
+            _ => {
+                info!("get segment attribute for record_size failed due to {:?}", reply);
```

Comment:
Humm, this actually brings up an interesting edge case:
Suppose the user creates the reader before the writer. Then we would get a segment does not exist error. In this case we have a couple of options: We could fail and simply not allow creating the reader first.
To make sure the fallback value is correct, we should add a check on reader:350 to make sure the value read was actually the correct size.